firmware/quadruna/VC/src/io/io_sbgEllipse.c

```diff
@@ -220,15 +221,50 @@ static void io_sbgEllipse_processMsg_EkfNavVelandPos(const SbgBinaryLogData *log
 
     // if (sbgEComLogEkfGetSolutionMode(log_data->ekfNavData.status) != SBG_ECOM_SOL_MODE_NAV_POSITION)
     // {
-    //     uint32_t status = log_data->ekfNavData.status;
+        // uint32_t status = log_data->ekfNavData.status;
 
-    //     bool is_velocity_valid = (status & SBG_ECOM_SOL_VELOCITY_VALID) != 0;
-    //     bool is_position_valid = (status & SBG_ECOM_SOL_POSITION_VALID) != 0;
+        // bool is_velocity_invalid = (status & SBG_ECOM_SOL_VELOCITY_VALID) != 0;
+        // bool is_position_valid = (status & SBG_ECOM_SOL_POSITION_VALID) != 0;
 
-    //     if (!is_velocity_valid & !is_position_valid)
-    //         return;
+        // bool is_data_invalid = !is_velocity_valid & !is_position_valid;
+
+        // if (is_data_invalid) {
+        // app_canTx_VC_Warning_VelocityDataInvalid_set(is_velocity_invalid);
+
+        // if (is_velocity_invalid)
+        // {
+            // app_canTx_VC_Warning_EllipseSolutionModeWarning_set(is_data_invalid);
+            // io_sbgEllipse_calculateVelocity();
+        //     return;
+        // }
     // }
 
+    uint32_t status = log_data->ekfNavData.status;
+
+    bool is_moving =
+        (app_canTx_VC_LeftInverterTorqueCommand_get() > 0.0f) || (app_canTx_VC_RightInverterSpeedCommand_get() > 0.0f);
+
+    if (is_moving)
```

Comment:
i would do this ismoving logic in app, u can do it inside app_broadcast and add it as a function there. I would add `uint32_t ekf_solution_status;` into sensor data and then make a getter function like 
`uint32_t io_sbgEllipse_getSolutionStatus(void)
{
    return sensor_data.ekf_solution_status;
}`

 In the io_sbgEllipse_processMsg_EkfNavVelandPos function delete all the changes including the ` app_canTx_VC_EkfSolutionMode_set((VcEkfStatus)sbgEComLogEkfGetSolutionMode(log_data->ekfNavData.status));`. in the app all states where u do the is moving function i would do it there. 
I suggest u add a counter later on to see how many times this has failed in a row before being like alright this is definitily not working let's just stop doing closed loop vehicle velocity functionality

Reply:
just use motor rpm get or smth